skills/c3-component-design/SKILL.md

```diff
@@ -25,403 +25,117 @@ Also called by c3-adopt to CREATE initial Component documentation.
-| Performance tuning | Specific optimizations | Pool sizing formula |
-| Health checks | Implementation | `SELECT 1` ping |
-| Testing approach | How to test | Mock strategies |
-
-**EXCLUDE from Component (push to Container or Context):**
-
-| Element | Why Not Component | Where It Belongs |
-|---------|------------------|------------------|
-| Container purpose | Too high level | Container |
-| API endpoint list | Container scope | Container |
-| Middleware order | Container pipeline | Container |
-| Technology choice rationale | Container decision | Container |
-| System protocols | System-wide | Context |
-| Cross-cutting concerns | Span containers | Context |
-| Deployment topology | System-wide | Context |
-
-### Litmus Test
-
-Ask: "Could a developer implement this from the documentation?"
-- **Yes** → Correct level (Component)
-- **No, needs more detail** → Add implementation specifics
-- **No, it's about structure** → Push up to Container
+## Core Principles for Component Documents
+- **Abstraction:** Implementation detail. Show HOW components work to satisfy container obligations.
+- **Nature-driven focus:** Nature tags are open-ended; choose what best explains behavior (resource, business, cross-cutting, entrypoint, testing, deployment, contextual, etc.).
+- **Anchors:** Use `{#com-xxx-*}` anchors for linkable sections.
+- **Derivation:** Components implement container protocols/cross-cutting; reference the container sections they fulfill.
+- **Environment ownership:** Component doc owns stack details and configuration (with env diffs).
 
 ---
 
-## Expressing Relationships at Component Level
+## Nature Types (open-ended, pick what fits)
 
-### Relationship Types
+| Nature | Focus |
+|--------|-------|
+| Resource/Integration | Configuration, env differences, how/why config loaded |
+| Business Logic | Domain flows, rules, edge cases |
+| Framework/Entrypoint | Mixed concerns: auth, errors, lifecycle |
+| Cross-cutting | Integration patterns and how used everywhere |
+| Build/Deployment | Build pipeline, deploy config, CI/CD specifics |
+| Testing | Strategy, fixtures, mocking approach |
+| Contextual | Caching, websocket, feature-specific behavior |
+| Other | Add any nature that clarifies the focus |
 
-| Relationship | Expression | Example |
-|--------------|------------|---------|
-| Dependency injection | Constructor param | `constructor(private db: Pool)` |
-| Method call | Interface method | `taskService.create(data)` |
-| Event emission | Event name + payload | `emit('task.created', { task })` |
-| Data flow | Input → Transform → Output | `request → validate → persist` |
-| Error propagation | Throws/catches | `throws DatabaseError` |
-
-### Interface Documentation
-
-```markdown
-## Service Interface
-
-```typescript
-interface TaskService {
-  create(userId: string, data: CreateInput): Promise<Task>;
-  update(taskId: string, data: UpdateInput): Promise<Task>;
-  delete(taskId: string): Promise<void>;
```

Comment:
The lean approach here (-368 lines) is interesting. My PR #3 went the opposite direction (+300 lines) by keeping more guidance/examples.

Question: With this minimal template, do we lose helpful context for newcomers, or is the intent that they'll learn from the examples directory instead?

The tradeoff seems to be: **skill file as reference doc** vs **skill file as minimal template**.